src/encoded/static/components/experiment_series.js

```diff
@@ -85,85 +291,71 @@ class ExperimentSeriesComponent extends React.Component {
-            // Retrieve the audits of all related datasets.
-            const datasetAtIds = relatedDatasets.map(dataset => dataset['@id']);
-            Promise.all(datasetAtIds.map(datasetAtId => (
-                fetch(`${datasetAtId}?frame=audit`, {
-                    method: 'GET',
-                    headers: {
-                        Accept: 'application/json',
-                    },
-                }).then((response) => {
-                    // Convert each response response to JSON
-                    if (response.ok) {
-                        return response.json();
-                    }
-                    return Promise.resolve(null);
-                })
-            ))).then((datasetAudits) => {
-                // All dataset audits have been retrieved, or we got an error when trying to
-                // retrieve one or more of them.
-                const auditsByDataset = {};
-                if (datasetAudits && datasetAudits.length > 0) {
-                    datasetAudits.forEach((datasetAudit) => {
-                        auditsByDataset[datasetAudit['@id']] = datasetAudit.audit || {};
-                    });
-                }
-                this.setState({ auditsByDataset });
-            });
-        }
+        this.getViewableDatasets();
     }
 
     /**
-     * Calculate a list of related_datasets that we can view given our access level and each
-     * dataset's status and place this list into this.viewableDatasets.
-     *
-     * @return {array} All datasets from ExperimentSeries object we can display. Empty array if
-     *                 none.
+     * Retrieve viewable datasets and set it to the viewableDatasets state.
+     * The viewableDatasets is based on context.related_datasets,
+     * filtered based on experiment status and added audit and quality metrics.
      */
     getViewableDatasets() {
-        if (this.props.context.related_datasets.length > 0) {
-            const accessLevel = sessionToAccessLevel(this.context.session, this.context.session_properties);
-            const viewableStatuses = getObjectStatuses('Dataset', accessLevel, viewableDatasetStatuses);
-            return this.props.context.related_datasets.filter(dataset => viewableStatuses.includes(dataset.status));
+        // Get viewable datasets
+        if (this.props.context.related_datasets.length === 0) {
+            return [];
         }
+        const accessLevel = sessionToAccessLevel(this.context.session, this.context.session_properties);
+        const viewableStatuses = getObjectStatuses('Dataset', accessLevel, viewableDatasetStatuses);
+        const viewableDatasets = this.props.context.related_datasets.reduce(
+            (datasetObjects, dataset) => {
+                if (viewableStatuses.includes(dataset.status)) {
+                    datasetObjects[dataset['@id']] = dataset;
+                }
+                return datasetObjects;
+            },
+            {}
+        );
```

Comment:
Simiarly to above, this doesn’t look like a reduce() operation. This is generating a filtered object keyed by dataset.@id which could be done in a `forEach`.